js/lesson-05.js

```diff
@@ -0,0 +1,125 @@
+// !!!  ДЗ 10. Масив
+// Створити масив, довжину та елементи якого задає користувач.
+// Відсортувати масив за зростанням.
+// Видалити елементи з масиву з 2 по 4 (включно!).
+// У міру змін виводити вміст масиву на сторінку.
+
+// !!! Рішення
+
+// const arrayLength = Number(prompt('Введіть довжину для масиву')); 
+// if (arrayLength === 0) {
+//    alert (`Ви не ввели довжину для масиву`)
+// } else if(isNaN(arrayLength)) {
+//    alert (`Ви ввели не числове значення для довжини масиву`)
+// } else {
+//    const userArray = new Array(arrayLength);
+//    console.log(userArray);
+   
+//    for (let i=0; i<arrayLength; i++) {
+//       userArray[i] = prompt('Введіть дані для масиву','');
+//    };
+//    console.log(`Ви ввели такі данні для масиву: ${userArray}`)
+   
+//    const sortedArray = userArray.sort();
+//    console.log(`Відсортовані данні для масиву: ${sortedArray}`)
+   
+//    const slicedArray = [];
+//    sortedArray.forEach((item, i) => {
+//       if(i < 1 || i > 3) {
+//          slicedArray.push(item);
+//       }
+//    })
+//    console.log(`Відформатовані данні для масиву: ${slicedArray}`)
+// }
+
+
+   
+   // !!!  ДЗ 11. Пошук у масиві
+   
+   // Даний масив [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+   
+
+   // ??? Знайти суму та кількість позитивних елементів.
+   // ???     Визначити кількість негативних елементів.
+   // ???       Знайти кількість непарних позитивних елементів.
+   // ???  Знайти кількість парних позитивних елементів.
+   // ???  Знайти мінімальний елемент масиву та його порядковий номер.
+   // ???   Знайти максимальний елемент масиву та його порядковий номер.
+   
+   const arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47];
+   let positiveNumbers = 0;
+   let negativeNumbers = 0;
+   let positiveOddNumbers = 0;
+   let positiveEvenNumbers = 0;
+   const minNumber = Math.min(...arr);
+   const minNumberIndex = arr.findIndex((number) => number === minNumber);
+   const maxNumber = Math.max(...arr);
+   const maxNumberIndex = arr.findIndex((number) => number === maxNumber);
+
+   const sum = arr.reduce((currentSum, number) => {
+      if (number > 0) {
+         if (number % 2 !== 0) {
+            positiveOddNumbers++;
+      } else {
+         positiveEvenNumbers++;
+      }
+      positiveNumbers++;
+      return (currentSum = currentSum + number);
```

Comment:
не треба зінювати параметри функціїї
Ваш код працює лише тому, що оператор присвоєння повертає значення яке було присвоєне
Пишіть отак:

```suggestion
      return currentSum + number;
```

Reply:
нижче ще є такі самі випадки